src/NeoSharp.Core/Network/Server.cs

```diff
@@ -10,60 +13,65 @@ public class Server : IServer
         // ReSharper disable once NotAccessedField.Local
         private readonly ILogger<Server> _logger;
         private readonly NetworkConfig _cfg;
-        private readonly Func<IPeer> _peerFactory;
-        
+        private readonly IPeerFactory _peerFactory;
+
         // ReSharper disable once FieldCanBeMadeReadOnly.Local
         // ReSharper disable once CollectionNeverUpdated.Local
-        private IList<IPeer> _connectedPeers;                // if we successfully connect with a peer it is inserted into this list
+        private ConcurrentBag<IPeer> _connectedPeers;                // if we successfully connect with a peer it is inserted into this list
         // ReSharper disable once NotAccessedField.Local
         private IList<IPEndPoint> _failedPeers;             // if we can't connect to a peer it is inserted into this list
         private uint _nonce;                                // uniquely identifies this server so we can filter out our own messages sent back to us by other nodes
```

Comment:
I prefer Guid for next version of protocol, is more unique, maybe i should add this change to https://github.com/neo-project/proposals/pull/31

Reply:
I am not familiar with the protocol yet. Will it make this incompatible with old one?